web-server/pkg/apis/k8sinitiative.3scale.net/v1alpha1/types.go

```diff
@@ -0,0 +1,26 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+//go:generate controller-gen object paths=$GOFILE
+
+type Product struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              ProductSpec `json:"spec"`
+}
+
+type ProductSpec struct {
```

Comment:
All attributes are required?

My advise, add all of them as optional. It is much easier for latter deprecation/updating...

´´´
Description   *string `json:"description,omitempty"`
´´´